contents/swift/welcome-to-swift.md

```diff
@@ -12,44 +12,47 @@
+Swift is a fast, modern and safe programming language mainly used for iOS programming.
+Introduced in 2014, Swift has concise and expressive syntax, especially when compared with its predecessor language **Objective-C**. On top of that, Swift also supports playgrounds, a feature that allows programmers to experiment with Swift code and see the results immediately (without the overhead of building and running an app).
 
-# Getting Started
+## Why should I learn Swift?
 
-Reading the [Language Guide in the official documentation](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/) is definitely recommended, since it explains everything quite clearly, albeit quite verbose.
+1. **Learning Swift can help you land your dream career.**
+Swift is mainly used for iOS development and iOS developers are well paid. The [salary statistics](https://www.indeed.com/salaries/iOS-Developer-Salaries) shows that iOS developer in US can earn more than $120K per year. In addition, mobile application development is also becoming more popular as of recent times. A [survey in StackOverFlow](https://insights.stackoverflow.com/survey/2017#developer-profile-developer-type) shows that mobile development, although lesser than web development, still takes up a large share of the market.
 
-If you're just looking to take a dip into Swift, here are the [basics](https://guides.codepath.com/ios/Swift-Basics) you'll need to understand.
+1. **Swift is easy to learn.** Swift is designed to be a simple and understandable language by Apple. Its syntax resembles English, and thus is easy to read and understand. Any programmer who is familiar with Python, Java or C++ can learn Swift quickly! Even if you are not familiar with any language, Swift is an excellent language to start out with.
 
-And if you are really pressed for time, here's a couple of cheatsheets with code examples:
+1. **Swift is an [open source project](https://github.com/apple/swift).** Learning about Swift and contributing to the language may be a useful experience for those interested. Apart from learning about Swift from its documentation, you can also dig into the source code to get new insights. Being open-source means you can also start submitting issues or patches if you find there is any way to improve the language!
 
-- [Swift Cheat Sheet](https://mhm5000.gitbooks.io/swift-cheat-sheet/content/index.html) (web-friendly)
-- [Swift 3.0 Cheat Sheet and Quick Reference](https://koenig-media.raywenderlich.com/uploads/2014/06/RW-Swift-Cheatsheet-0_7.pdf) (print-friendly)
+Also note that Swift is not perfect. It is still very young, and therefore the language itself may have issues to be addressed, such as [stored properties in extension](https://medium.com/@valv0/computed-properties-and-extensions-a-pure-swift-approach-64733768112c).
```

Comment:
What is the purpose of bringing this point here? 

It would be nice if you can bring this in a tip box and say something like 'Swift is still a young language and there are plenty of opportunities for you to improve it' 

Put it in a box like this:
```
<box type="tip">
....
</box>
```

Reply:
Oh, the [suggested structure](https://se-education.org/learningresources/contributing.html) says that try to give the balanced view about the topic. i.e. mention both advantages and disadvantages. Therefore, I think it's fine to keep this part in the main context, and "there are plenty of opportunities for you to improve it" is good to add

Reply:
Yup I see where your coming from!  

It is up to you to put it a box. I am fine with either way.

Reply:
Probably leave this as a separate section called "limitations" at the end of this article.

Reply:
I am not sure whether it's better to put it along with advantages or in a separate section. I just follow the suggested structure to put them together